climada/util/config.py

```diff
@@ -321,3 +321,33 @@ def _fetch_conf(directories, config_name):
     Path(Path.home(), '.config'),  # ~/.config directory
     Path.cwd(),  # current working directory
 ], CONFIG_NAME))
+
+
+class log_level:
+    """Context manager that sets ALL loggers to a given level."""
+
+    def __init__(self, level, climada_only=True):
+        self.level = level
+        self.loggers = {
+            name: logger
+            for name, logger in logging.root.manager.loggerDict.items()
+            if isinstance(logger, logging.Logger)
+            }
+        if climada_only:
+            self.loggers = {
+                name: logger
+                for name, logger in self.loggers.items()
+                if 'climada' in name
+                }
+            
+        self.loglevels = [logger.level
+            for logger in self.loggers.values()
+            ] 
+
+    def __enter__(self):
+        for logger in self.loggers.values():
+            logger.setLevel(self.level)
+
+    def __exit__(self, exception_type, exception, traceback):
+        for logger, level in zip(self.loggers.values(), self.loglevels):
+            logger.setLevel(level)
```

Comment:
I suggest to have a kwarg for the logger prefix that defaults to an empty string, then filter by that name. Furthermore, the previous log-levels shouldn't be stored in a list because the order of dict keys/values might be non-persistent per the official specification.
```suggestion
    def __init__(self, level, name_prefix=""):
        self.level = level
        self.loggers = {
            name: (logger, logger.level)
            for name, logger in logging.root.manager.loggerDict.items()
            if isinstance(logger, logging.Logger) and name.startswith(name_prefix)
            }
        if name_prefix == "":
            self.loggers[""] = logging.getLogger()

    def __enter__(self):
        for logger, _ in self.loggers.values():
            logger.setLevel(self.level)

    def __exit__(self, exception_type, exception, traceback):
        for logger, previous_level in self.loggers.values():
            logger.setLevel(previous_level)
```

Reply:
Good catch again! Should we allow only for prefix selection or generally any keyword present anywhere in the logger name?